### PR TITLE
Create new user api

### PR DESCRIPTION
**Pull request by AndreyStadnyk**
Added userDTO, creating new user api, userpassword crypting

### CODE REVIEW

File: src/main/java/ua/com/danit/controller/UserController.java [hunk 1]
```diff
@@ -2,35 +2,48 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/controller/UserController.java]: Returrn rows with all imports, please.

File: src/main/java/ua/com/danit/controller/UserController.java [hunk 2]
```diff
@@ -2,35 +2,48 @@
...
+import ua.com.danit.dto.request.PostRequest;
+import ua.com.danit.dto.request.UserRequest;
+import ua.com.danit.dto.response.UserResponse;
 import ua.com.danit.entity.User;
+import ua.com.danit.mapping.UserMapper;
 import ua.com.danit.service.UserService;
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/controller/UserController.java]: Delete this import, please.

File: src/main/java/ua/com/danit/dto/request/UserRequest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public class UserRequest {
+
+  private String username;
+  private String firstName;
+  private String lastName;
+  private LocalDate birthDate;
+  private String address;
+  private String gender;
+  private String password;
+  private List<Post> posts = new ArrayList<>();
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/dto/request/UserRequest.java]: Maybe, it's must be List of PostRequest
And for rows below too.

File: src/main/java/ua/com/danit/dto/response/CommentResponse.java [hunk 4]
```diff
@@ -4,8 +4,8 @@
 
 @Data
 public class CommentResponse {
-  public long id;
+  public Long id;
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/dto/response/CommentResponse.java]: I think primitive types is better for Response entity

File: src/main/java/ua/com/danit/dto/response/UserResponse.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+@Data
+public class UserResponse {
+
+  private String username;
+  private String firstName;
+  private String lastName;
+  private LocalDate birthDate;
+  private String address;
+  private String gender;
+  private List<Post> posts = new ArrayList<>();
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/dto/response/UserResponse.java]: Maybe, it's must be List of PostResponse
And for rows below too.

File: src/main/java/ua/com/danit/controller/UserController.java [hunk 6]
```diff
@@ -2,35 +2,46 @@
...
-import ua.com.danit.entity.User;
-import ua.com.danit.service.UserService;
-
-import java.util.List;
+import ua.com.danit.dto.request.UserRequest;
+import ua.com.danit.dto.response.UserResponse;
+import ua.com.danit.mapping.UserMapper;
 
 @RestController
 @RequestMapping("/api/v1/users")
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/controller/UserController.java]: We don't set v1 in route

File: src/main/java/ua/com/danit/entity/User.java [hunk 7]
```diff
@@ -2,15 +2,7 @@
...
-import javax.persistence.CascadeType;
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.JoinTable;
-import javax.persistence.ManyToMany;
-import javax.persistence.OneToMany;
-import javax.persistence.Table;
+import javax.persistence.*;
```
Comment (alex5272372) [inline src/main/java/ua/com/danit/entity/User.java]: Incorrect import

File: src/main/resources/logback.xml [hunk 8]
```diff
@@ -1,9 +1,9 @@
...
-  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
-    <encoder>
-      <pattern>%d{HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</pattern>
-    </encoder>
-  </appender>
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <encoder>
+            <pattern>%d{HH:mm:ss} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
```
Comment (alex5272372) [inline src/main/resources/logback.xml:6]: Discard these changes. Indent must be 2 spaces

**Review by alex5272372 — APPROVED** [review]

**Review by ShevchukMarina — APPROVED** [review]

**Pull request merged**